### PR TITLE
Per paper, make potentials signed (instead of {0,1}) and IMPLY natural

### PR DESCRIPTION
**Pull request by alldefector**
See section 2.4 of http://i.stanford.edu/hazy/papers/inc.pdf

I didn't touch `FUNC_LINEAR`, `FUNC_RATIO`, and `FUNC_LOGICAL` because they seem to be experiment code and the grounding setup seems different from the typical representation.

Related to https://github.com/HazyResearch/sampler/issues/26

### CODE REVIEW

**Comment by alldefector** [comment]
This change will make it more intuitive to emulate bool vars with categorical vars (which we may actually need to do soon). In the coin flipping setting, the two weights for a two-value categorical would be same absolute value with opposite signs.

File: test/sampler_test.cc [hunk 1]
```diff
@@ -53,13 +53,13 @@ TEST_F(SamplerTest, sample_sgd_single_variable) {
   sampler->set_random_seed(1, 1, 1);
 
   sampler->sample_sgd_single_variable(0, 0.1);
-  EXPECT_EQ(infrs->weight_values[0], 0.1);
+  EXPECT_EQ(infrs->weight_values[0], 0.2);
```
Comment (netj) [inline test/sampler_test.cc]: Do you think the stepsize should get adjusted or it doesn't matter?

**Comment by netj** [comment]
LGTM  I can't think of any sophisticated downside for now.  Someone reusing old weights can get surprised by new results though.

File: test/sampler_test.cc (hunk 1 above)
Comment (alldefector) [inline test/sampler_test.cc]: The default step size is `0.01`, and I assume that was at least somewhat arbitrary... The default decay is 0.95, so after about 13 epochs, `0.02` would become `0.01` anyway...

**Pull request merged**